website/blog/2022-12-13-pointer-events-in-react-native.md

```diff
@@ -0,0 +1,175 @@
+| onPointerOver  | onPointerCancel\* | onClick               |
+| onPointerEnter |                   | onContextMenu         |
+| onPointerDown  |                   | onGotPointerCapture   |
+| onPointerMove  |                   | onLostPointerCapture  |
+| onPointerUp    |                   | onPointerRawUpdate    |
+| onPointerOut   |                   |                       |
+| onPointerLeave |                   |                       |
+
+<!-- prettier-ignore -->
+<sub>* onPointerCancel has been hooked up to the native platform’s “cancel” event but this does not necessarily correspond to when the web platform expects them to fire.</sub>
+
+### Event Properties
+
+For each of the events mentioned above we’ve also implemented the majority of the properties expected in the PointerEvent object — though in React Native these are exposed through the `event.nativeEvent` property. You can find an enumeration of all the implemented properties in the [event object’s Flowtype interface definition](https://github.com/facebook/react-native/blob/59ee57352738f030b41589a450209e51e44bbb06/Libraries/Types/CoreEventTypes.js#L175). One notable exception to being completely implemented is the `relatedTarget` property as exposing a native view reference in this ad-hoc manner isn’t trivial.
+
+## Future Work and Exploration
+
+In addition to the events above there are also some other APIs related to Pointer Events. In the future, we plan to be implement these APIs as a part of this effort. These APIs include:
+
+- Pointer Capture API
+  - Includes the imperative API exposed on element references including `setPointerCapture()`, `releasePointerCapture()`, and `hasPointerCapture()`.
+- `touch-action` style property
+  - The web uses this CSS property to declaratively negotiate gestures between the browser and a website’s own event handling code. In React Native this could be used for negotiating the event handling between a View’s pointer event handlers and a parent ScrollView.
+- `click`, `contextmenu`, `auxclick`
+  - `click` is an abstract definition of interaction that may be triggered through accessibility paradigms or other characteristic platform interactions.
+
+Another benefit of the native Pointer Events implementation is that it will allow us to revisit and improve various forms of gesture handling currently limited only to touch events and handled in JavaScript by the Responder, Pressability, and PanResponder APIs.
+
+Furthermore, we are continuing to explore including an implementation of the `EventTarget` interface for React Native host components (i.e. `add`/`removeEventListener`) which we believe will make possible more user-land abstractions for handling pointer interactions.
+
+## Trying it Out
+
+Our Pointer Events implementation is still experimental but we’re interested in getting feedback from the community on what we’ve shared. If you are interested in trying this API you’ll need to enable a couple feature flags:
```

Comment:
Maybe worth clarifying that there isn't yet support on desktop, since we mention it as an inspiration for this work.

Reply:
Though I guess that's also kind of implied by saying it requires Fabric.